### PR TITLE
Update deploy a smart contract page

### LINKED ISSUE #1575: Update the Deploy a Smart Contract page
**Describe the solution you'd like**

1. The current "npx hardhat" command does not include the "Create a basic sample project" option as used on [this page](https://wiki.iota.org/isc/how-tos/deploy-a-smart-contract/#requirements). Instead, it presents the following options:

- Create a JavaScript project
- Create a TypeScript project
- Create a TypeScript project (with Viem)
- Create an empty hardhat.config.js

The instruction to "Select 'Create a basic sample project'" should be replaced with one of the above options in step 5 of the process.

2. Additionally, it would be beneficial to include instructions on obtaining YOUR PRIVATE KEY, as demonstrated on [this page](https://wiki.iota.org/tutorials/shimmerevm-testnet-hardhat/).

3. The current `deploy.js` script produces the following error:

`TypeError: counter.deployed is not a function`

Fix:

Commenting out `await counter.deployed()` allows the contract to be deployed, but to an undefined address. Using `await counter.getAddress()` in the console log statement correctly displays the deployed address. We should replace `counter.address` with `await counter.getAddress()` to properly display the address.

### PR DESCRIPTION
**Pull request by Ginowine**
# Description of change

This PR made some update to this [page on the wiki](https://wiki.iota.org/isc/how-tos/deploy-a-smart-contract/). The PR modifies the deploy.js script, and also added how to get your private key.

## Links to any relevant issues

fixes #1575  

## Type of change

- Documentation Enhancement

## Change checklist

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work

### CODE REVIEW

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 1]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
...
+
+- Click "Export Private Key"
+
+- To access your private key, you will now need to enter your wallet password. When done, click Confirm to continue.
+
+- Your private key will now be displayed. Click to copy it and save it in a safe place. (Note: For obvious reasons, we didn't show it in the screenshots below - but yours will be there.)
+
+- Click close icon on top to close the screen.
+
+Follow [this article](https://helpwithpenny.com/export-and-import-metamask-private-key/) for further detail.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:168]:
```suggestion
You can find more information in the [official Metamask Documentation](https://support.metamask.io/managing-my-wallet/secret-recovery-phrase-and-private-keys/how-to-export-an-accounts-private-key/).
```

Comment (lucas-tortora) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:168]: Please use official docs only for external references.

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 2]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
 
 <HardhatConfig/>
 
+:::info Private Key
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:150]:
```suggestion
:::info Export the Metamask Private Key
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 3]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
...
+
+How to export the MetaMask private key of an account?
+
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:153]:
```suggestion
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 4]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
...
+- Click on the logo in the upper right corner
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:154]:
```suggestion
1. Click on the Metamask logo in the upper right corner.
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 5]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
...
+
+- Select the account you want to export
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:156]:
```suggestion
2. Select the account you want to export.
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 6]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
...
+
+- On the account page, click the menu (three dots) in the upper right corner, then click the "Account Details" button:
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:158]:
```suggestion
3. On the account page, click the menu (three dots) in the upper right corner, then click the "Account Details" button.
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 7]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
...
+
+- Click "Export Private Key"
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:160]:
```suggestion
4. Click on "Export Private Key".
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 8]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
...
+
+- To access your private key, you will now need to enter your wallet password. When done, click Confirm to continue.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:162]:
```suggestion
5. Enter your wallet password to access your private key and click `Confirm` to continue.
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 9]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
...
+
+- Your private key will now be displayed. Click to copy it and save it in a safe place. (Note: For obvious reasons, we didn't show it in the screenshots below - but yours will be there.)
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:164]:
```suggestion
6. Your private key will now be displayed. Click to copy it and save it in a safe place. 
```

Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:164]:
```suggestion
6. Your private key will now be displayed. Click to copy it and save it in a safe place. 
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 10]
```diff
@@ -149,6 +147,28 @@ project. Here's a step-by-step guide:
...
+
+- Click close icon on top to close the screen.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:166]:
```suggestion
```
This seems unnecessary.

**Review by lucas-tortora — CHANGES_REQUESTED** [review]
Just some small comments about style. Btw, you can create branches directly in this repo, no need for forks :)

**Comment by Ginowine** [comment]
Thanks, I'll stop using my forked repo. working on your suggestion now.

**Review by lucas-tortora — APPROVED** [review]

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx [hunk 11]
```diff
@@ -109,10 +109,10 @@ project. Here's a step-by-step guide:
    ```
 5. Create a Hardhat Project by running the following command:
    ```bash
-   npx hardhat
+   npx hardhat init
    ```
 
-   Select "Create a basic sample project" when prompted and answer the setup questions (you can just press enter to
+   Select "Create a JavaScript project" when prompted and answer the setup questions (you can just press enter to
```
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/deploy-a-smart-contract.mdx:115]:
```suggestion
   Select `Create a JavaScript project` (or whatever applies to your project) when prompted and answer the setup questions (you can press enter to
```

**Review by Dr-Electron — APPROVED** [review]

**Pull request merged**